spec/chartmogul/opportunity_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe ChartMogul::Opportunity do
+  let(:attrs) do
+    {
+      uuid: uuid,
+      customer_uuid: customer_uuid,
+      owner: 'kamil+pavlicko@chartmogul.com',
+      pipeline: 'New Business',
+      pipeline_stage: 'Discovery',
+      estimated_close_date: '2024-03-30',
+      currency: 'USD',
+      amount_in_cents: 100_000,
+      type: 'one-time',
+      forecast_category: 'best_case',
+      win_likelihood: 30,
+      custom: [{ key: 'from_campaign', value: true }]
+    }
+  end
+
+  let(:uuid) { '3726cd86-e0f0-11ee-8d22-93b1d57bf35d' }
+  let(:customer_uuid) { 'cus_8bae29d0-df51-11ee-88c8-97dc7855258a' }
+  let(:updated_attributes) { { estimated_close_date: '2024-04-30' } }
+  let(:cursor) { 'MjAyNC0wMy0xMVQwNjozODo0Ny44Nzk2NDYwMDBaJjAzZDdkNWJjLWRmNzItMTFlZS1iMTBkLThiNzg5ZDc1N2MyYQ==' }
+
+  describe '#initialize' do
+    subject { described_class.new(attrs) }
+
+    it 'sets the read-only properties correctly' do
+      expect(subject).to have_attributes({ uuid: nil })
+    end
+
+    it 'sets the writeable properties correctly' do
+      expect(subject).to have_attributes(attrs.reject { |k, _| k == :uuid })
+    end
+  end
```

Comment:
[minor] we can add more tests for `create_at` and `updated_at`

Reply:
added 👍